src/server/v2.0/handler/label.go

```diff
@@ -0,0 +1,218 @@
+		label.ProjectID = 0
+	case common.LabelScopeProject:
+		exist, err := lAPI.projectCtl.Exists(ctx, label.ProjectID)
+		if err != nil {
+			return lAPI.SendError(ctx, errors.Wrapf(err, "failed to check the existence of project %d", label.ProjectID))
+		}
+		if !exist {
+			return lAPI.SendError(ctx, errors.New(nil).WithMessage("project %d not found", label.ProjectID).WithCode(errors.BadRequestCode))
+		}
+	}
+
+	if err := lAPI.requireAccess(ctx, label, rbac.ActionCreate); err != nil {
+		return lAPI.SendError(ctx, err)
+	}
+
+	query := q.New(q.KeyWords{"Name": label.Name, "Level": label.Level, "Scope": label.Scope, "ProjectID": label.ProjectID})
+
+	labels, err := lAPI.labelMgr.List(ctx, query)
+	if err != nil {
+		return lAPI.SendError(ctx, err)
+	}
+	if len(labels) > 0 {
+		return lAPI.SendError(ctx, errors.New("conflict label").WithCode(errors.ConflictCode))
+	}
+
+	id, err := lAPI.labelMgr.Create(ctx, label)
+	if err != nil {
+		return lAPI.SendError(ctx, err)
+	}
+
+	location := fmt.Sprintf("%s/%d", strings.TrimSuffix(params.HTTPRequest.URL.Path, "/"), id)
+	return operation.NewCreateLabelCreated().WithLocation(location)
+}
+
+func (lAPI *labelAPI) GetLabelByID(ctx context.Context, params operation.GetLabelByIDParams) middleware.Responder {
+	label, err := lAPI.labelMgr.Get(ctx, params.LabelID)
+	if err != nil {
+		return lAPI.SendError(ctx, err)
+	}
+	if label == nil || label.Deleted {
+		return lAPI.SendError(ctx, errors.New(nil).WithMessage("label %d not found", params.LabelID).WithCode(errors.NotFoundCode))
+	}
+
+	if err := lAPI.requireAccess(ctx, label, rbac.ActionRead); err != nil {
+		return lAPI.SendError(ctx, err)
+	}
+
+	return operation.NewGetLabelByIDOK().WithPayload(model.NewLabel(label).ToSwagger())
+}
+
+func (lAPI *labelAPI) ListLabels(ctx context.Context, params operation.ListLabelsParams) middleware.Responder {
+	query, err := lAPI.BuildQuery(ctx, params.Q, params.Sort, params.Page, params.PageSize)
+	if err != nil {
+		return lAPI.SendError(ctx, err)
+	}
+
+	scope := lib.StringValue(params.Scope)
+	if scope != common.LabelScopeGlobal && scope != common.LabelScopeProject {
+		return lAPI.SendError(ctx, errors.New(nil).WithMessage("invalid scope: %s", scope).WithCode(errors.BadRequestCode))
+	}
+	query.Keywords["Level"] = common.LabelLevelUser
+	query.Keywords["Scope"] = scope
+	name := lib.StringValue(params.Name)
+	if name != "" {
+		query.Keywords["name"] = &q.FuzzyMatchValue{Value: name}
+	}
+	if scope == common.LabelScopeProject {
+		pid := lib.Int64Value(params.ProjectID)
+		if pid == 0 {
+			return lAPI.SendError(ctx, errors.BadRequestError(nil).WithMessage("must with project ID when to query project labels"))
+		}
+		if err := lAPI.RequireProjectAccess(ctx, pid, rbac.ActionList, rbac.ResourceLabel); err != nil {
+			return lAPI.SendError(ctx, err)
+		}
+		query.Keywords["ProjectID"] = pid
+	}
+
+	total, err := lAPI.labelMgr.Count(ctx, query)
+	if err != nil {
+		return lAPI.SendError(ctx, err)
+	}
```

Comment:
Return directly when the total is zero to reduce a DB query.

Reply:
updated